### ISSUE TITLE
field name compact

### ISSUE THREAD

**Issue by jkindle** [msgpack/msgpack#171]
could msgpack add map field name compact feature by define a global field name array?
for example:
{"longlongfieldname":"testvalue"} -> {"$0":"testvalue"}  by define a array var keywords = ["longlongfieldname"]

**Comment by kuenishi** [comment]
I don't think this should be done in binary representation level, but with on top of MessagePack object level.

**Comment by lowsfer** [comment]
I also thought about it recently, because my data contains a long array of struct, so I guess the field names of the struct would be repeated many times in msgpack, which is inefficient.

While it's possible to implement it on top of MessagePack, that's more cumbersome compared to an schema-based format, which typically use integers as keys and covers this optimization automatically. I wonder if it's possible to have a global mode flag in the message to provide this optimization, then use integer keys and include an a global list of de-duplicated field names.
